src/eos-updater-poll-common.c

```diff
@@ -0,0 +1,910 @@
+                       const gchar *ref,
+                       const gchar *url_override,
+                       gchar **out_checksum,
+                       EosExtensions **out_extensions,
+                       GError **error)
+{
+  if (commit_checksum_from_extensions_ref (repo,
+                                           cancellable,
+                                           remote_name,
+                                           ref,
+                                           url_override,
+                                           out_checksum,
+                                           out_extensions,
+                                           error))
+    return TRUE;
+
+  g_clear_error (error);
+  if (commit_checksum_from_extensions_summary (repo,
+                                               cancellable,
+                                               remote_name,
+                                               ref,
+                                               url_override,
+                                               out_checksum,
+                                               out_extensions,
+                                               error))
+    return TRUE;
+
+  g_clear_error (error);
+  if (commit_checksum_from_summary (repo,
+                                    cancellable,
+                                    remote_name,
+                                    ref,
+                                    url_override,
+                                    out_checksum,
+                                    out_extensions,
+                                    error))
+    return TRUE;
+
+  g_clear_error (error);
+  if (url_override != NULL)
+    g_set_error (error, G_IO_ERROR, G_IO_ERROR_FAILED,
+                 "Failed to get the checksum of the latest commit in ref %s from remote %s with URL %s",
+                 ref, remote_name, url_override);
+  else
+    g_set_error (error, G_IO_ERROR, G_IO_ERROR_FAILED,
+                 "Failed to get the checksum of the latest commit in ref %s from remote %s",
+                 ref, remote_name);
+  return FALSE;
+}
+
+gboolean
+fetch_latest_commit (OstreeRepo *repo,
+                     GCancellable *cancellable,
+                     const gchar *remote_name,
+                     const gchar *ref,
+                     const gchar *url_override,
+                     gchar **out_checksum,
+                     EosExtensions **out_extensions,
+                     GError **error)
+{
+  g_autoptr(GBytes) summary_bytes = NULL;
+  g_autoptr(GVariant) summary = NULL;
+  g_autofree gchar *checksum = NULL;
+
+  g_return_val_if_fail (OSTREE_IS_REPO (repo), FALSE);
+  g_return_val_if_fail (cancellable == NULL || G_IS_CANCELLABLE (cancellable), FALSE);
+  g_return_val_if_fail (remote_name != NULL, FALSE);
+  g_return_val_if_fail (ref != NULL, FALSE);
+  g_return_val_if_fail (out_checksum != NULL, FALSE);
+
+  {
+    g_autoptr(GVariant) options = NULL;
```

Comment:
Is there any reason to put `options` in a separate scope to the rest of this function?